### ISSUE TITLE
[Request] `ros-noetic-desktop-full` Robot Operating System (ROS)

### ISSUE THREAD

**Issue by acxz** [arch4edu/arch4edu#78]
Now that `ros-noetic` has been released and is working on the AUR, I thought it would be a good time to open this package request issue for `ros-noetic-desktop-full`. This is the successor to #44

**Comment by acxz** [comment]
@petronny would it be possible to get a rebuild for all the ros-noetic-* packages? Some people have been facing issues using the arch4edu ros packages.

**Comment by petronny** [comment]
Could you give me a link to the issues?

**Comment by acxz** [comment]
Yes that is exactly the case.

**Comment by petronny** [comment]
> we haven't traced down which ROS packages to be rebuilt

I think there is a way to do this. But I don't have time to implement it.
I just leave the psudeo-code here (in a python-like grammar):
```python
for package_i in all_ros_packages:
  so_files = find `.so*` files in package_i
  for so_file in so_files:
    linked_dynamic_libraries = get the linked dynamic libraries of so_file by `ldd`
    for linked_dynamic_library in linked_dynamic_libraries:
      if linked_dynamic_library matches `.so.pkgver*`:
        rebuild_dependency = find which package contains linked_dynamic_library by `pacman -F`
        print(f"{package_i} should be rebuilt when {rebuild_dependency} is updated.")
```

**Comment by AlexandrParkhomenko** [comment]
ros-noetic-desktop-full is fine build again

**Comment by gh7531** [comment]
What is the current state of the ros-noetic-desktop-full package? The ROS install guide as well as the arch wiki link to this repository but I can't find it.

**Comment by AlexandrParkhomenko** [comment]
> The ROS install guide as well as the arch wiki link to this repository but I can't find it.

[Install on archlinux is simply.](https://github.com/aur-archlinux/ros) Just use [yay](https://aur.archlinux.org/packages/yay)

**Comment by petronny** [comment]
I'll start packaging these packages until all the other packages are built.

PRs are welcome if you want to package them now.

**Comment by acxz** [comment]
Closing this issue as ros2 will soon be preferred over `ros-noetic`

**Comment by acxz** [comment]
I would like to give an update this task, I just recently was able to build all of ros-noetic-desktop-full in a clean chroot. This should make packaging efforts much easier now. This has been possible due to `paru` having devtools support.

However, the issue with automated rebuilding with newer dependency versions has still not be resolved. I don't feel comfortable asking @petronny to continue packaging the ros-noetic packages until that issue is properly resolved.

And as I mentioned earlier, I have plans to start distributing `ros-rolling` (i.e. ros2) soon. As even now, I can start feeling the effects of having to backport fixes for ros-noetic packages, similar to what happened to ros-melodic, before we transitioned to ros-noetic

ps: glad to hear the bot is running again!
And hope you having been keeping well @petronny
